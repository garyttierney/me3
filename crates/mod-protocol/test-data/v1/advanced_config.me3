profileVersion = "v1"
disable_arxan = false
start_online = true

[[supports]]
game = "nightreign"

[[packages]]
id = "my-mod"
source = "./mod"

[[packages]]
source = "./unnamed-mod"
load_after = [{ id = "my-mod", optional = false }]

[[packages]]
id = "my-other-mod"
path = "./nr-mods/mod"
enabled = true
optional = true
load_before = [{ id = "my-mod", optional = true }]

[[packages]]
id = "my-disabled-mod"
source = "./unused-mod"
enabled = false

[[natives]]
path = "my_native.dll"
optional = true

[[natives]]
path = "./nr-mods/my_other_native.dll"
initializer.function = "init_my_dll"
