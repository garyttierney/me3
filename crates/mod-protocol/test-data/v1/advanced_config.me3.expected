V1(
    ModProfileV1 {
        supports: [
            Supports {
                game: Nightreign,
                since_version: None,
            },
        ],
        natives: [
            Native {
                inner: ModFile {
                    name: "my_native",
                    path: "my_native.dll",
                    enabled: true,
                    optional: true,
                },
                initializer: None,
                load_before: [],
                load_after: [],
            },
            Native {
                inner: ModFile {
                    name: "my_other_native",
                    path: "./nr-mods/my_other_native.dll",
                    enabled: true,
                    optional: false,
                },
                initializer: Some(
                    NativeInitializerCondition {
                        delay: None,
                        function: Some(
                            "init_my_dll",
                        ),
                    },
                ),
                load_before: [],
                load_after: [],
            },
        ],
        packages: [
            Package {
                inner: ModFile {
                    name: "my-mod",
                    path: "./mod",
                    enabled: true,
                    optional: false,
                },
                load_before: [],
                load_after: [],
            },
            Package {
                inner: ModFile {
                    name: "unnamed-mod",
                    path: "./unnamed-mod",
                    enabled: true,
                    optional: false,
                },
                load_before: [],
                load_after: [
                    Dependent {
                        id: "my-mod",
                        optional: false,
                    },
                ],
            },
            Package {
                inner: ModFile {
                    name: "my-other-mod",
                    path: "./nr-mods/mod",
                    enabled: true,
                    optional: false,
                },
                load_before: [
                    Dependent {
                        id: "my-mod",
                        optional: true,
                    },
                ],
                load_after: [],
            },
            Package {
                inner: ModFile {
                    name: "my-disabled-mod",
                    path: "./unused-mod",
                    enabled: false,
                    optional: false,
                },
                load_before: [],
                load_after: [],
            },
        ],
        savefile: None,
        start_online: Some(
            true,
        ),
        disable_arxan: Some(
            false,
        ),
    },
)
